public/app/plugins/datasource/cloud-monitoring/types.ts

```diff
@@ -63,33 +58,72 @@ export enum EditorMode {
   MQL = 'mql',
 }
 
-export const queryTypes = [
-  { label: 'Metrics', value: QueryType.METRICS },
-  { label: 'Service Level Objectives (SLO)', value: QueryType.SLO },
-];
+export enum PreprocessorType {
+  None = 'none',
+  Rate = 'rate',
+  Delta = 'delta',
+}
 
-export interface MetricQuery {
-  editorMode: EditorMode;
+export enum MetricKind {
```

Comment:
Because these kind of get added to the global auto-complete namespace, I've been manually namespacing the Azure types with AzureBlah to help me find the right import I want. just something to consider.

Reply:
interesting, haven't thought about that.